format.js

```diff
@@ -0,0 +1,58 @@
+// consider this a warning about getting obsessive about optimization
+
+var utilformat = require('util').format
+
+
+function format (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
```

Comment:
It *shouldn't* so long as you don't do re-assigns on named arguments...

We [do this in core](https://github.com/nodejs/node/blob/master/lib/timers.js#L383-L407) and I feel like you might make it back by using the switch since it might optimize better.

Also, keeping this under 50 LOC so that it is inline-able will definitely help.